packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -3334,8 +3448,65 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
       config.hint = hint;
     if (textDirection != null)
       config.textDirection = textDirection;
-    if (button != null)
-      config.isButton = button;
+    // Registering _perform* as action handlers instead of the user provided
+    // ones to ensure that changing a user provided handler from a non-null to
+    // another non-null value doesn't require a semantics update.
+    if (onTap != null)
+      config.addAction(SemanticsAction.tap, _performTap);
+    if (onLongPress != null)
+      config.addAction(SemanticsAction.longPress, _performLongPress);
+    if (onScrollLeft != null)
+      config.addAction(SemanticsAction.scrollLeft, _performScrollLeft);
+    if (onScrollRight != null)
+      config.addAction(SemanticsAction.scrollRight, _performScrollRight);
+    if (onScrollUp != null)
+      config.addAction(SemanticsAction.scrollUp, _performScrollUp);
+    if (onScrollDown != null)
+      config.addAction(SemanticsAction.scrollDown, _performScrollDown);
+    if (onIncrease != null)
+      config.addAction(SemanticsAction.increase, _performIncrease);
+    if (onDecrease != null)
+      config.addAction(SemanticsAction.decrease, _performDecrease);
+  }
+
+  void _performTap() {
+    if (onTap != null)
```

Comment:
i guess the concern is that the system will invoke our callbacks sometime after we've been told we no longer have an onTap handler?

Reply:
That was my concern. Does the system guarantee that that will never happen?

Reply:
At the render level, I guess not. You can modify a render object, then drive the pipeline much later.